### PR TITLE
News um Meta-Description und Meta-Titel erweitern (SEO)

### PR DESCRIPTION
**Pull request by eS-IT**
This PR implements #87.

### CODE REVIEW

File: news-bundle/src/Resources/contao/classes/News.php [hunk 1]
```diff
@@ -494,6 +494,37 @@ public function purgeOldFeeds()
...
+	 * @param PageRegular $objPageRegular
+	 */
+	public function setMetaData(PageModel $objPage, LayoutModel $objLayout, PageRegular $objPageRegular)
+    {
+        if (!\Input::get('items'))
+        {
+            return; // no news
+        }
+
+        $news = \NewsModel::findOneBy(array('alias=?', 'published=?'), array($this->Input->get('items'), 1));
```
Comment (Toflar) [inline news-bundle/src/Resources/contao/classes/News.php]: You can use `\Input::get('items')` here again, or even better, store it in a variable on line 506 :)

File: news-bundle/src/Resources/contao/languages/de/tl_news.xlf [hunk 2]
```diff
@@ -25,6 +25,22 @@
         <source>Here you can change the author of the news item.</source>
         <target>Hier können Sie den Autor des Beitrags ändern.</target>
       </trans-unit>
+      <trans-unit id="tl_news.metatitle.0">
```
Comment (Toflar) [inline news-bundle/src/Resources/contao/languages/de/tl_news.xlf]: You only need to add English, the rest is managed on Transifex :)

**Review by Toflar — COMMENTED** [review]
Two minor comments, excellent work otherwise :)

File: news-bundle/src/Resources/contao/languages/de/tl_news.xlf [hunk 3]
```diff
@@ -327,4 +327,4 @@
       </trans-unit>
     </body>
   </file>
-</xliff>
```
Comment (Toflar) [inline news-bundle/src/Resources/contao/languages/de/tl_news.xlf]: This file does not need to be adjusted :) (you've deleted the final new line. Our coding style forces to always end a file with a new line :)

File: news-bundle/src/Resources/contao/config/config.php [hunk 4]
```diff
@@ -46,6 +46,7 @@
 $GLOBALS['TL_HOOKS']['generatePage'][] = array('contao_news.listener.generate_page', 'onGeneratePage');
 $GLOBALS['TL_HOOKS']['generateXmlFiles'][] = array('News', 'generateFeeds');
 $GLOBALS['TL_HOOKS']['replaceInsertTags'][] = array('contao_news.listener.insert_tags', 'onReplaceInsertTags');
+$GLOBALS["TL_HOOKS"]['generatePage'][] = array('News', 'setMetaData');
```
Comment (leofeyer) [inline news-bundle/src/Resources/contao/config/config.php]: We already have a listener for the `generatePage` hook three lines above. Wouldn't it make sense to re-use it?

**Comment by leofeyer** [comment]
Here's what needs to be done IMHO:

- [ ] Re-use the existing listener
- [ ] Rename the fields from `metatitle` to `title` and from `metadescription` to `description`, so they match `tl_page.title` and `tl_page.description`
- [ ] Put the fields into a separate palette analogous to `tl_page`

**Comment by eS-IT** [comment]
In `tl_page` heißt das Feld `pageTitle`, da das Feld `title` der " Seitenname" ist. Soll es in `tl_news` `title` oder `pageTitle` heißen?

**Review by ausi — CHANGES_REQUESTED** [review]
IMO `GeneratePageListener::setMetaData()` is missing a check if we are actually on a news page. If I got a calendar event with the same alias as a news entry, wouldn’t this code set the title/description for the news on the event page then?

Shouldn’t the code for setting the title/description instead go into the reader module as we already set the page title there? https://github.com/contao/contao/blob/578044764b17db7cbec78c94d0beae589dc6dbd6/news-bundle/src/Resources/contao/modules/ModuleNewsReader.php#L105

**Comment by leofeyer** [comment]
> Shouldn’t the code for setting the title/description instead go into the reader module as we already set the page title there?

@ausi You are absolutely right.

**Comment by leofeyer** [comment]
> In `tl_page` heißt das Feld `pageTitle`, da das Feld `title` der " Seitenname" ist. Soll es in `tl_news` `title` oder `pageTitle` heißen?

Gute Frage. Ich denke `tl_news.pageTitle`, weil das zumindest konsistent wäre.

**Comment by contaoacademy** [comment]
Vielen herzlichen Dank, dass Ihr das Feature so schnell implementiert.

**Comment by leofeyer** [comment]
Thank you @eS-IT.

**Pull request merged**